### PR TITLE
Tma 528 kbv txma integration

### PR DESCRIPTION
**Pull request by dthampi**
Created a feature test to test the integration from KBV to TxMA in staging env

### CODE REVIEW

File: tests/audit-tests/src/test/resources/features/FirehoseToS3.feature [hunk 1]
```diff
@@ -1,8 +1,8 @@
-@build @dev @staging
-Feature: IPV event data journey from firehose to S3
-
-  Scenario: Verify the data journey from firehose to S3
-    Given the SNS file "firehose" is available
-    And the expected file "s3_expected" is available
-    When the message is sent to firehose
-    Then the audit s3 should have a new event matching the output file
\ No newline at end of file
+#@build @dev @staging
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/resources/features/FirehoseToS3.feature:1]: These should be uncommented

File: tests/audit-tests/src/test/resources/features/kbv_Txma.feature [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    Given the user is on KBV CRI Staging
+    When the user searches Kenneth Decerqueira and click `Search`
+    When the user clicks on `Go to KBV CRI Staging`
+    When the user answers the first question correctly
+    When the user answers the second KBV question correctly
+    When the user answers the third KBV question correctly
+    Then the verifiable credential page should be displayed
+    Then Response from Address CRI Integration displays the user's address in JSON
+    When the user clicks on summaryTest and reads the sub value from JSON
+#    Then the audit S3 should have a new event with the postcode provided
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/resources/features/kbv_Txma.feature:15]: Should be uncommented and should be searching for the user_id

File: tests/audit-tests/src/test/resources/features/kbv_Txma.feature [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  @staging
+  Scenario: KBV to TxMA integration test in Staging
+    Given the user is on KBV CRI Staging
+    When the user searches Kenneth Decerqueira and click `Search`
+    When the user clicks on `Go to KBV CRI Staging`
+    When the user answers the first question correctly
+    When the user answers the second KBV question correctly
+    When the user answers the third KBV question correctly
+    Then the verifiable credential page should be displayed
+    Then Response from Address CRI Integration displays the user's address in JSON
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/resources/features/kbv_Txma.feature:13]: Should be named something generic as used in both features

File: tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/kbv_TxmaStepDefinitions.java [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+
+        }
+    }
+
+    @Then("the verifiable credential page should be displayed")
+    public void the_verifiable_credential_page_should_be_displayed() throws JsonProcessingException {
+        BrowserUtils.waitForPageToLoad(100);
+        new VerifiableCredentialsPage().ResponseFromKBVCRIStaging.click();
+        String ResponseJSON = new VerifiableCredentialsPage().Data.getText();
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/kbv_TxmaStepDefinitions.java:535]: These two lines don't do anything (I don't think anyway)

File: tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/kbv_TxmaStepDefinitions.java [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+        JsonNode jsonNode = objectMapper.readTree(ResponseJSON);
+        JsonNode vcNode = jsonNode.get("vc");
+        JsonNode evidenceNode = vcNode.get("evidence");
+        JsonNode insideEvidence = evidenceNode.get(0);
+        String verificationScore = insideEvidence.get("verificationScore").asText();
+        System.out.println("verificationScore = " + verificationScore);
+    }
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/kbv_TxmaStepDefinitions.java:551]:
```suggestion
```
This isn't used

File: tests/audit-tests/src/test/java/utilities/kbv/BrowserUtils.java [hunk 6]
```diff
@@ -0,0 +1,387 @@
+package utilities.kbv;
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/kbv/BrowserUtils.java:1]: None of this BrowserUtils is used

File: tests/audit-tests/src/test/java/utilities/kbv/ConfigurationReader.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+ * reads the properties file configuration.properties
+ */
+public class ConfigurationReader {
+
+    private static Properties properties;
+
+    static {
+
+        try {
+            String path = "configuration.properties";
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/kbv/ConfigurationReader.java:17]: Why can't I see this?

File: tests/audit-tests/src/test/java/utilities/kbv/ConfigurationReader.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+    public static String get(String keyName) {
+        return properties.getProperty(keyName);
+    }
+
+    public static String getBrowser() {
+        return Optional.ofNullable(System.getenv("BROWSER")).orElse("chrome-headless");
+    }
+
+    public static String getVisitCredentialIssuersUrl() {
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/kbv/ConfigurationReader.java:37]: The rest of this page is not used

File: tests/audit-tests/src/test/java/utilities/kbv/ConfigurationReader.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
+package utilities.kbv;
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/kbv/ConfigurationReader.java:1]: Can you just use the other Configuration Reader?

File: tests/audit-tests/src/test/java/utilities/kbv/Driver.java [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        }
+        return driverPool.get();
+    }
+
+    public static void closeDriver() {
+        driverPool.get().quit();
+        driverPool.remove();
+        WebDriver driver=new ChromeDriver();
+        driver.close(); //closes the browser
+    }
```
Comment (Aidan23Smith) [inline tests/audit-tests/src/test/java/utilities/kbv/Driver.java:93]: This is the only difference between this driver and the other one, and it's not used

**Review by PeterBrownGDS — APPROVED** [review]

File: tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/kbv_TxmaStepDefinitions.java [hunk 11]
```diff
@@ -0,0 +1,552 @@
...
+                } catch (Exception e) {
+                    if (new HowMuchOfYourLoanDoYouHaveLeftToPayBackPage().OVER6500UPTO6750.isDisplayed()) {
+                        new HowMuchOfYourLoanDoYouHaveLeftToPayBackPage().OVER6500UPTO6750.click();
+                        new HowMuchOfYourLoanDoYouHaveLeftToPayBackPage().Continue.click();
+                    }
+                }
+                break;
+            case "What are the first two letters of the forename of the other person on your mortgage? – Prove your identity – GOV.UK":
+                new WhatAreTheFirstTwoLettersOfTheForenameOfTheOtherPersonOnYourMortgagePage().KA.click();
+                new WhatAreTheFirstTwoLettersOfTheForenameOfTheOtherPersonOnYourMortgagePage().Continue.click();
```
[the remark this answers is not in the document]
Comment (PeterBrownGDS) [reply tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/kbv_TxmaStepDefinitions.java]: Lets go with yes

**Review by PeterBrownGDS — DISMISSED** [review]

**Review by Aidan23Smith — APPROVED** [review]
Looks good to me (based on Pete's approval)

**Pull request merged**